test/SlowTests/Server/Replication/PullReplicationPreventDeletionsTests.cs

```diff
@@ -0,0 +1,395 @@
+            {
+                Database = hubStore.Database, Name = hubStore.Database + "ConStr", TopologyDiscoveryUrls = hubStore.Urls
+            }));
+
+            await sinkStore.Maintenance.SendAsync(new UpdatePullReplicationAsSinkOperation(new PullReplicationAsSink
+            {
+                ConnectionStringName = hubStore.Database + "ConStr",
+                Mode = PullReplicationMode.SinkToHub | PullReplicationMode.HubToSink,
+                CertificateWithPrivateKey = Convert.ToBase64String(pullCert.Export(X509ContentType.Pfx)),
+                HubName = "pullRepHub"
+            }));
+
+            using (var s = sinkStore.OpenAsyncSession())
+            {
+                dynamic user1 = new {Source = "Sink"};
+                await s.StoreAsync(user1, "users/insink/1");
+                s.Advanced.GetMetadataFor(user1)[Constants.Documents.Metadata.Expires] = DateTime.UtcNow.AddMinutes(10);
+
+                dynamic user2 = new {Source = "Sink"};
+                await s.StoreAsync(user2, "users/insink/2");
+                s.Advanced.GetMetadataFor(user2)[Constants.Documents.Metadata.Expires] = DateTime.UtcNow.AddMinutes(10);
+
+                await s.SaveChangesAsync();
+            }
+
+            using (var s = hubStore.OpenAsyncSession())
+            {
+                await s.StoreAsync(new {Source = "Hub"}, "users/inhub/1");
+                await s.SaveChangesAsync();
+            }
+
+            Assert.True(WaitForDocument(sinkStore, "users/inhub/1"));
+
+            //make sure hub got both docs and expires doesn't get deleted
+            using (var h = hubStore.OpenAsyncSession())
+            {
+                //check hub got both docs
+                var doc1 = await h.LoadAsync<dynamic>("users/insink/1");
+                Assert.NotNull(doc1);
+
+                var doc2 = await h.LoadAsync<dynamic>("users/insink/2");
+                Assert.NotNull(doc2);
+
+                //check expired exists in users/insink/1
+                var metadata = h.Advanced.GetMetadataFor(doc1);
+                var expires = metadata[Constants.Documents.Metadata.Expires];
+                Assert.NotNull(expires);
+
+                //check expired exists in users/insink/2
+                metadata = h.Advanced.GetMetadataFor(doc2);
+                expires = metadata[Constants.Documents.Metadata.Expires];
+                Assert.NotNull(expires);
+            }
+
+            //delete doc from sink
+            using (var s = sinkStore.OpenAsyncSession())
+            {
+                s.Delete("users/insink/1");
+                await s.SaveChangesAsync();
+            }
+
+            EnsureReplicating(hubStore, sinkStore);
+            EnsureReplicating(sinkStore, hubStore);
+
+            //make sure doc is deleted from hub and sink both
+            Assert.False(WaitForDocument(hubStore, "users/insink/1"));
+            Assert.False(WaitForDocument(sinkStore, "users/insink/1"));
```

Comment:
This is wrong, you are needlessly waiting.
You want to go the other way around, Wait for the document to NOT be there.

Reply:
Done